pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -287,3 +294,34 @@ func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, vn
 
 	return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided vnet '%s' is invalid: custom DNS servers are not supported.", vnetID)
 }
+
+func (dv *openShiftClusterDynamicValidator) validateProviders(ctx context.Context, providerClient resources.ProvidersClient) error {
+	providers := []string{
+		"Microsoft.Storage",
+		"Microsoft.Authorization",
+		"Microsoft.Compute",
+		"Microsoft.Network",
+	}
+
+	providersList, err := providerClient.List(ctx, nil, "")
+	if err != nil {
+		return err
+	}
+	for _, rp := range providersList {
+		found := false
+		for _, provider := range providers {
+			if provider == *rp.Namespace {
+				if *rp.RegistrationState != "Registered" {
+					return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidResourceProviderPermissions, "", "The resource provider '%s' is not registered.", *rp.Namespace)
+				}
+				found = true
+				break
+			}
+		}
+		if !found {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidResourceProviderPermissions, "", "The resource provider '%s' is missing.", *rp.Namespace)
```

Comment:
I added an error here when the provider is not returned by the query to list providers.  Is this what is expected?  @jim-minter